File: src/main/java/com/oshi/libgenericsettings/data/TitleSubtitleExtraData.java
package com.oshi.libgenericsettings.data;

import android.support.annotation.ColorRes;

/**
 * Created by udioshi on 9/4/2017.
 */
public class TitleSubtitleExtraData extends TitleSubtitleData {

    protected String extraText;
    protected @ColorRes int extraColor;

    public TitleSubtitleExtraData(String title, String subtitle, String extraText) {
        super(title, subtitle);
        this.extraText = extraText;
    }

    public TitleSubtitleExtraData(String extraText, @ColorRes int extraColor) {
        this.extraText = extraText;
        this.extraColor = extraColor;
    }

    public TitleSubtitleExtraData(String title, String subtitle, String extraText, @ColorRes int extraColor) {
        super(title, subtitle);
        this.extraText = extraText;
        this.extraColor = extraColor;
    }

    public TitleSubtitleExtraData(String title, String subtitle, @ColorRes int titleColor, @ColorRes int subtitleColor, String extraText, @ColorRes int extraColor) {
        super(title, subtitle, titleColor, subtitleColor);
        this.extraText = extraText;
        this.extraColor = extraColor;
    }

    public String getExtraText() {
        return extraText;
    }

    public void setExtraText(String extraText) {
        this.extraText = extraText;
    }

    public int getExtraColor() {
        return extraColor;
    }

    public void setExtraColor(@ColorRes int extraColor) {
        this.extraColor = extraColor;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_TITLE_SUBTITLE_EXTRA;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/TitleSubtitleExtraData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes
import com.oshi.libgenericsettings.data.BaseViewTypeData.Companion.SettingsViewTypes

/**
 * Settings Title, Subtitle & an Extra text. Being use in layout/view_type_title_subtitle_extra.xml
 */
class TitleSubtitleExtraData(title : String, subtitle : String, var extraText : String) : TitleSubtitleData(title, subtitle) {

    @ColorRes var extraColor : Int = 0

    override fun getViewType(): SettingsViewTypes {
        return SettingsViewTypes.VIEW_TYPE_SETTINGS_TITLE_SUBTITLE_EXTRA
    }
}